### ISSUE TITLE
[Compat] PHP 8.1

### ISSUE THREAD

**Issue by Gounlaf** [mcustiel/phiremock-common#12]
Hi @mcustiel,

Long time haven't used Phiremock.
Today, I got the following warning in a PHP 8.1 environment:

```
Deprecated: Return type of Mcustiel\Phiremock\Domain\AbstractArrayIterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /opt/app/vendor/mcustiel/phiremock-common/src/Domain/AbstractArrayIterator.php on line 60

Deprecated: Return type of Mcustiel\Phiremock\Domain\AbstractArrayIterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /opt/app/vendor/mcustiel/phiremock-common/src/Domain/AbstractArrayIterator.php on line 38

Deprecated: Return type of Mcustiel\Phiremock\Domain\AbstractArrayIterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /opt/app/vendor/mcustiel/phiremock-common/src/Domain/AbstractArrayIterator.php on line 80

Deprecated: Return type of Mcustiel\Phiremock\Domain\AbstractArrayIterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /opt/app/vendor/mcustiel/phiremock-common/src/Domain/AbstractArrayIterator.php on line 48

Deprecated: Return type of Mcustiel\Phiremock\Domain\AbstractArrayIterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /opt/app/vendor/mcustiel/phiremock-common/src/Domain/AbstractArrayIterator.php on line 70

Deprecated: Return type of Mcustiel\Phiremock\Domain\AbstractArrayIterator::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /opt/app/vendor/mcustiel/phiremock-common/src/Domain/AbstractArrayIterator.php on line 90
```

Could you update the library for compatiblity? Or do you need a PR?

Please let me know.

Regards.

**Comment by mcustiel** [comment]
Hey @Gounlaf Glad to hear you're back. I fixed it in PR #13 
Thanks for reporting.

**Comment by Gounlaf** [comment]
Thanks! 👌
